resolwe/flow/managers/listener/listener.py

```diff
@@ -69,22 +69,155 @@
 logger = logging.getLogger(__name__)
 
 
-class Processor:
-    """Class respresents instance of one connected worker."""
+class RedisCache:
+    """The Redis cache."""
+
+    cached_fields_type = {
+        "id": int,
+        "slug": str,
+        "contributor": int,
+        "contributor_id": int,
+        "status": str,
+        "started": datetime,
+        "finished": datetime,
+        "process": int,
+        "process_id": int,
+        "process__slug": str,
+        "process_error": "liststr",
+        "process_warning": "liststr",
+        "process_info": "liststr",
+        "descriptor_schema": int,
+        "descriptor_schema_id": int,
+        "tags": list,
+        "process_memory": int,
+        "process_cores": int,
+        "location": int,
+        "location_id": int,
+        # Assuming this is dict with integer values.
+        # This may change in the future.
+        "process_resources": "dictint",
+        "worker": int,
+        "worker_id": int,
+        "worker__status": str,
+    }
+    cached_fields_set = set(cached_fields_type)
+
+    def __init__(self, redis: redis.Redis, *args, **kwargs):
+        """Set the cached field set."""
+        # The set of fields on the data object that can be safely cached inside
+        # Redis to avoid hitting the database (we are assuming Redis is faster).
+        # The list contains a set of commonly used fields that do not change
+        # from the outside (or do no harm if they do).
+
+        self._redis = redis
+
+        super().__init__(*args, **kwargs)
+
+    def get_redis_key(self, data_id: int, field_name: str) -> str:
+        """Get the redis key from the field name."""
+        return f"resolwe-listener-data-{data_id}-redis-cache-{field_name}"
+
+    def get_cache(
+        self,
+        data_id: int,
+        field_names: Iterable[str],
+    ) -> Dict[str, Any]:
+        """Obtain the set of fields from the redis cache.
 
-    def __init__(
+        If field_name is requested that is not cached the request is silently
+        ignored.
+
+        The query is run in a transaction.
+        """
+
+        def get_redis_data(pipeline: redis.client.Pipeline):
+            def get_redis_field(redis_key: str, field_name: str) -> Any:
+                """Get the redis data from the specific key."""
+                field_type = self.cached_fields_type[field_name]
+
+                if field_type in (int, str, datetime):
```

Comment:
It would be nice to create a wrapper class for each field tye and move CRUD operations there.

Reply:
I think the code is simple and it serves the scope for which it is used. If the cache will be used for more than just caching some data values than I agree.